src/log/file_appender.cpp

```diff
@@ -31,19 +37,46 @@ namespace fc {
              return time_point_sec( (uint32_t)(file_number * interval_seconds) );
          }
 
+         void setup_log_filename_values()
+         {
+            _now = time_point::now();
+            _start_time = get_file_start_time( _now, cfg.rotation_interval );
+            _timestamp_string = _start_time.to_non_delimited_iso_string();
+            _link_filename = cfg.filename;
+            _log_filename = _link_filename.parent_path() / (_link_filename.filename().string() + "." + _timestamp_string);
+         }
+
+         void open_log_file()
+         {
+            // on windows, you can't delete the link while the underlying file is opened for writing
+            remove_all(_link_filename);
+            out.open( _log_filename, std::ios_base::out | std::ios_base::app );
+            create_hard_link(_log_filename, _link_filename);
+         }
+
       public:
          impl( const config& c) : cfg( c )
          {
-             if( cfg.rotate )
-             {
-                 FC_ASSERT( cfg.rotation_interval >= seconds( 1 ) );
-                 FC_ASSERT( cfg.rotation_limit >= cfg.rotation_interval );
+            setup_log_filename_values();
 
+            try
+            {
+               fc::create_directories(cfg.filename.parent_path());
 
+               open_log_file();
+            }
+            catch( ... )
+            {
+               std::cerr << "error opening log file: " << cfg.filename.preferred_string() << "\n";
```

Comment:
Is the appender usable after caught an exception here? What will happen when trying to write?